service/s3/internal/customizations/presign_test.go

```diff
@@ -0,0 +1,180 @@
+package customizations_test
+
+import (
+	"bytes"
+	"context"
+	"net/http"
+	"strings"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/internal/awstesting/unit"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+)
+
+func TestPutObject_PresignURL(t *testing.T) {
+	cases := map[string]struct {
+		input                   s3.PutObjectInput
+		options                 s3.PresignOptions
+		expectPresignedURLHost  string
+		expectedRequestURILabel []string
+		expectSignedHeader      http.Header
+		expectMethod            string
+		expectError             string
+	}{
+		"standard case": {
```

Comment:
One additional testcase that would be nice to have would be one where the user explicitly specifies a `X-Amz-Content-Sha256` header value the request should be presigned with.

Reply:
That value should be used for the presigned URL, and would be a signed header that must be sent by the client using the presigned URL.